src/net.c

```diff
@@ -753,6 +760,82 @@ static int hyper_set_interface_mtu(struct rtnl_handle *rth,
 				IFLA_MTU);
 }
 
+static int hyper_get_iface_name_from_mac_addr(char *mac_addr, char **device)
+{
+	struct ifaddrs *ifaddr, *ifa;
+	struct ifreq ifr;
+	int sock;
+	char tmp_mac_addr[MAC_ADDR_STR_SIZE];
+
+	if (getifaddrs(&ifaddr) == -1) {
+		fprintf(stderr, "failed to get interface list: %s\n",
+			strerror(errno));
+		return -1;
+	}
+
+	sock = socket(PF_INET, SOCK_DGRAM, 0);
+	if (sock == -1) {
+		fprintf(stderr, "failed to get socket handle: %s\n",
+			strerror(errno));
+		goto err;
+	}
+
+	*device = NULL;
+
+	for (ifa = ifaddr; ifa != NULL; ifa = ifa->ifa_next) {
+		if (ifa->ifa_addr == NULL) {
+			continue;
+		}
+
+		if (!strncpy(ifr.ifr_name, ifa->ifa_name,
+			    strlen(ifa->ifa_name) + 1)) {
+			fprintf(stderr, "strncpy failed to copy interface"
+				" name: %s\n", strerror(errno));
+			goto err;
```

Comment:
No no, the done flag is for myself so that I know I did the change :). I will push soon